Update cardano-ledger-specs dependency

nfrisby opened the pull request:
This is a single commit. See the message.

Jasagredo commented on ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:
```diff
@@ -30,25 +30,24 @@ import           GHC.Records
+import           Cardano.Ledger.Allegra.Translation ()
+import qualified Cardano.Ledger.Core as Core
 import           Cardano.Ledger.Era (Crypto, SupportsSegWit (..))
+import qualified Cardano.Ledger.Era as SL (TranslateEra (..), TxInBlock)
```
`Cardano.Ledger.Era` was qualified above as `Core`. Maybe we should keep the same consistent naming, but also it could be the case that this module specifically deals with stuff very specific to `Cardano.Ledger.Era`. In that case I would maybe qualify it as `Era`?

nfrisby replied:
Ah; I have commit on the next PR that does that. I'll move it to this one.

I went with `.Era` because generally what we're getting from there is the same conceptual thing as we get from `.Core` --- even the module header on `.Era` says it exports "core" stuff.

nfrisby replied:
> Maybe it would be worth modifying other files to make the qualification more uniform across our code.

That is for Issue #3105, which I' don't want to address yet. I needed to rename them in the modules in this PR to avoid import conflicts, so I did it where necessary. There are several other modules with out-of-date imports.

We'll need to do a broader review and pick a Pareto-optimal scheme -- there are unfortunately multiple dimensions that come to mind.

Jasagredo replied:
Perfect! resolving this thread then.

Jasagredo reviewed (approved):
Except from that optional comment, the rest looks fine. However I did not look into the non-edited files to search for more occurrences of similar stuff. Maybe it would be worth modifying other files to make the qualification more uniform across our code.

nfrisby commented:
I pushed up a commit that disables the failing tests. See Issue #3134.

However, we might be able to fix up ledger in the next couple hours. So I'll wait until then -- maybe we can merge without this workaround.

nfrisby commented:
@Jasagredo I re-requested review, because I added the `Coherent` newtype and adjusted some instances.

The new `Arbitrary (Coherent (ShelleyBlock era))` instance in `Test.Consensus.Shelley.Generators` is the key change. Jared rapidly added a new generator to `cardano-ledger-specs` that is more careful than the (recently changed) `Arbitrary (SL.Block era)` instance that has supported our existing `Arbitrary (ShelleyBlock era)` instance. So now the properties we have that actually expect some integrity to the arbitrary Shelley blocks explicitly requires it via the `Coherent` wrapper in their arguments. (A `suchThat` or QC implication would have been hopelessly improbable.)

Jasagredo reviewed (approved):
Looks reasonable. `Coherent` looks right and the uses seem correct. I'm fine with merging this.

Jasagredo commented:
The Hydra build errors seem to be due to the long compiling times on `Ouroboros.Consensus.Cardano.Node`. Perhaps we should turn off those optimizations while this bug is present.

EncodePanda reviewed (approved):
Thank you for well described commit message. It was straight-forward to do the review (even after 3 days-off :) ).
In general I under the need for the `Coherent` wrapper. I get why some instances were wrapped, though I'm not sure why some other instances were wrapped. But I assume that was because GHC required it :)

nfrisby commented on ouroboros-consensus-mock-test/src/Test/Consensus/Ledger/Mock/Generators.hs:
```diff
@@ -61,11 +61,20 @@ instance Arbitrary (HeaderHash blk) => Arbitrary (Point blk) where
+    let hdr = mkSimpleHeader encode simpleStdHeader ext
+    return $ SimpleBlock hdr body
+
+-- | This blindly creates random values, so the block will not be valid, but
```
I updated the comment a bit and added a corresponding one on the other instance. Hope that helps.

nfrisby commented:
b117726b4 is b29f6e190 with comments/stylish-haskell changes only.

nfrisby commented:
I created a local variant that removes `Coherent` et al and only uses `genCoherentBlock` for `Arbitrary (ShelleyBlock era)`. I built that and am running the tests suites again locally. If that has roughly the same times as b117726 , then I'll merge this PR as-is. If not, I'll force-push the one without `Coherent` and merge that.

nfrisby commented:
The timings seem the same. Merging.

nfrisby commented:
1d5b0c7a5 is b117726 rebased onto `origin/master`

nfrisby commented:
bors r+
